fabric-tree-api-v1/src/main/java/net/fabricmc/fabric/api/tree/v1/FabricTreeRegistry.java

```diff
@@ -0,0 +1,125 @@
+package net.fabricmc.fabric.api.tree.v1;
+
+import com.mojang.serialization.Codec;
+import net.fabricmc.fabric.mixin.tree.BlockStateProviderTypeInvoker;
+import net.fabricmc.fabric.mixin.tree.FoliagePlacerTypeInvoker;
+import net.fabricmc.fabric.mixin.tree.TreeDecoratorTypeInvoker;
+import net.fabricmc.fabric.mixin.tree.TrunkPlacerTypeInvoker;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.math.intprovider.IntProvider;
+import net.minecraft.util.math.intprovider.IntProviderType;
+import net.minecraft.world.gen.foliage.FoliagePlacer;
+import net.minecraft.world.gen.foliage.FoliagePlacerType;
+import net.minecraft.world.gen.stateprovider.BlockStateProvider;
+import net.minecraft.world.gen.stateprovider.BlockStateProviderType;
+import net.minecraft.world.gen.treedecorator.TreeDecorator;
+import net.minecraft.world.gen.treedecorator.TreeDecoratorType;
+import net.minecraft.world.gen.trunk.TrunkPlacer;
+import net.minecraft.world.gen.trunk.TrunkPlacerType;
+
+/**
+ * <p>
+ * 	   A registry for {@link TrunkPlacerType}s, {@link FoliagePlacerType}s, {@link TreeDecoratorType}s,
+ * 	   {@link BlockStateProviderType}s and {@link IntProviderType}s.
+ * </p>
+ * <p>
+ *     Example for {@link TrunkPlacerType}s:<br>
+ *
+ *     <code>
+ *         public static final TrunkPlacerType<MyTrunkPlacer> MY_TRUNK_PLACER =	FabricTreeRegistry.registerTrunkPlacerType(new Identifier("example", "my_trunk_placer"), MyTrunkPlacer.CODEC);
+ *     </code>
+ * </p>
+ * <p>
+ *     Example for {@link FoliagePlacerType}s:<br>
+ *
+ *     <code>
+ *         public static final FoliagePlacerType<MyFoliagePlacer> MY_FOLIAGE_PLACER = FabricTreeRegistry.registerFoliagePlacerType(new Identifier("tutorial", "my_foliage_placer"), MyFoliagePlacer.CODEC);
+ *     </code>
+ * </p>
+ * <p>
+ *     Example for {@link TreeDecoratorType}s:<br>
+ *
+ *     <code>
+ *         public static final TreeDecoratorType<MyTreeDecorator> MY_TREE_DECORATOR = FabricTreeRegistry.registerTreeDecoratorType(new Identifier("tutorial", "my_tree_decorator"), MyTreeDecorator.CODEC);
+ *     </code>
+ * </p>
+ * <p>
+ *     Example for {@link BlockStateProviderType}s:<br>
+ *
+ *     <code>
+ *         public static final BlockStateProviderType<MyBlockStateProvider> MY_BLOCK_STATE_PROVIDER = FabricTreeRegistry.registerBlockStateProviderType(new Identifier("tutorial", "my_block_state_provider"), MyBlockStateProvider.CODEC);
+ *     </code>
+ * </p>
+ * <p>
+ *     Example for {@link IntProviderType}s:<br>
+ *
+ *     <code>
+ *         public static final IntProviderType<MyIntProvider> MY_INT_PROVIDER = FabricTreeRegistry.registerIntProviderType(new Identifier("tutorial", "my_int_provider"), MyIntProvider.CODEC);
+ *     </code>
+ * </p>
```

Comment:
I think it's better to put these examples under their individual methods.

Reply:
Okay, I'll do that. Thanks!